### PR TITLE
DSR-1416: Fix to create VERSION file and handle docker build from command line

### PR DESCRIPTION
**Pull request by BinduRao2018**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
In the current version, the VERSION file required by dockerfile isn't generated properly. This results in failure during initialization when the device service is started
Issue Number: 

## What is the new behavior?
Updated Makefile to generate VERSION file, to use it in Dockerfile to build docker images locally.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
-NA-

## Are there any specific instructions or things that should be known prior to reviewing?
No

## Other information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ The repository can be found on git at [libmraa](https://github.com/intel-iot-dev
 2. Build a docker image by using the following command
 ```
 sh> cd device-grove-c
+sh> make version 
 sh> docker build . -t < device-grove-c > -f ./scripts/Dockerfile.alpine-3.11
```
Comment (iain-anderson) [inline README.md]: should the docker build line not read simply
`sh> docker build . -t device-grove-c -f ./scripts/Dockerfile.alpine-3.11`

**Review by iain-anderson — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (BinduRao2018) [reply README.md]: <> - This is to indicate the image name can be anything and not fixed to device-grove-c. I can remove if it is not clear.

Comment (iain-anderson) [reply README.md]: Yes please, maybe just say underneath "replacing `device-grove-c` with your preferred image name if necessary" or something

**Review by iain-anderson — APPROVED** [review]
LGTM

**Pull request merged**